desc/objectives/linear_objectives.py

```diff
@@ -236,6 +236,170 @@ def update_target(self, thing):
             self._unjit()
 
 
+class ShareParameters(_Objective):
+    """Fix specific degrees of freedom to be the same between Optimizable things.
+
+    Parameters
+    ----------
+    things : list of Optimizable
+        list of objects whose degrees of freedom are being fixed to eachother's values.
+        Must be at least length 2, but may be of arbitrary length.
+        Every object must be of the same type, and have the same size array for the
```

Comment:
yea but then I thought of all the ways this could mess up, like `Equilibrium.R_lmn` and `FourierRZToroidalSurface.R_lmn` are the same name, yet even if they are the same shape it makes no sense to have them be shared.

I can change it if you think that is not a good enough counterpoint not to, but is there an example where it makes sense to allow this?